chore: update `prettier-plugin-astro` to match latest version

### Describe the bug

Currently `prettier-plugin-astro` is on [`0.14.1`](https://github.com/withastro/prettier-plugin-astro/releases/tag/v0.14.1), but peer dependency requires `^0.13.0`. 

When prettier plugin is installed before `@antfu/eslint-config`, the latest version is installed, which make dependencies conflict.

```console
❯ npm i
npm ERR! code ERESOLVE
npm ERR! ERESOLVE could not resolve
npm ERR! 
npm ERR! While resolving: @antfu/eslint-config@3.10.0
npm ERR! Found: prettier-plugin-astro@0.14.1
npm ERR! node_modules/prettier-plugin-astro
npm ERR!   dev prettier-plugin-astro@"^0.14.1" from the root project
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peerOptional prettier-plugin-astro@"^0.13.0" from @antfu/eslint-config@3.10.0
npm ERR! node_modules/@antfu/eslint-config
npm ERR!   dev @antfu/eslint-config@"^3.10.0" from the root project
npm ERR! 
npm ERR! Conflicting peer dependency: prettier-plugin-astro@0.13.0
npm ERR! node_modules/prettier-plugin-astro
npm ERR!   peerOptional prettier-plugin-astro@"^0.13.0" from @antfu/eslint-config@3.10.0
npm ERR!   node_modules/@antfu/eslint-config
npm ERR!     dev @antfu/eslint-config@"^3.10.0" from the root project
npm ERR! 
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
```

Should I downgrade my `prettier-plugin-astro` to `^0.13.0`?  
Maybe should we update the peer dependency to `^0.14.0` or `>=0.14.0`?

I would love to make PR, just let me know which one (`^` or `>=`) should be used.

### Reproduction

https://stackblitz.com/edit/github-vgi4kk?file=package.json

### System Info

```Shell
System:
    OS: Windows 10 10.0.19045
    CPU: (8) x64 Intel(R) Core(TM) i7-6700K CPU @ 4.00GHz
    Memory: 5.26 GB / 15.96 GB
  Binaries:
    Node: 22.11.0 - ~\AppData\Local\fnm_multishells\17816_1732655335845\node.EXE
    Yarn: 1.22.5 - ~\AppData\Roaming\npm\yarn.CMD
    npm: 10.9.0 - ~\AppData\Local\fnm_multishells\17816_1732655335845\npm.CMD
    pnpm: 8.15.3 - ~\AppData\Roaming\npm\pnpm.CMD
    bun: 1.1.29 - ~\.bun\bin\bun.EXE
  Browsers:
    Chrome: 131.0.6778.86
    Edge: Chromium (127.0.2651.74)
    Internet Explorer: 11.0.19041.4355
```


### Used Package Manager

npm

### Validations

- [X] Follow our [Code of Conduct](https://github.com/antfu/.github/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guide](https://github.com/antfu/contribute).
- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A, please open a GitHub Discussion instead.
- [X] The provided reproduction is a [minimal reproducible](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

### Contributions

- [X] I am willing to submit a PR to fix this issue
- [ ] I am willing to submit a PR with failing tests (actually just go ahead and do it, thanks!)

--- a/package.json
+++ b/package.json
@@ -50,7 +50,7 @@
     "eslint-plugin-react-refresh": "^0.4.4",
     "eslint-plugin-solid": "^0.14.3",
     "eslint-plugin-svelte": ">=2.35.1",
-    "prettier-plugin-astro": "^0.13.0",
+    "prettier-plugin-astro": "^0.14.0",
     "prettier-plugin-slidev": "^1.0.5",
     "svelte-eslint-parser": ">=0.37.0"
   },

